subsys/bluetooth/host_extensions/radio_notification_conn_cb.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/hci.h>
+#include <bluetooth/radio_notification_cb.h>
+#include <helpers/nrfx_gppi.h>
+#include <bluetooth/hci_vs_sdc.h>
+
+#include <nrfx_egu.h>
+
+#if defined(CONFIG_BT_RADIO_NOTIFICATION_CONN_CB_EGU_INST_EGU0)
+#define EGU_INST_IDX 0
+#elif defined(CONFIG_BT_RADIO_NOTIFICATION_CONN_CB_EGU_INST_EGU1)
+#define EGU_INST_IDX 1
+#elif defined(CONFIG_BT_RADIO_NOTIFICATION_CONN_CB_EGU_INST_EGU2)
+#define EGU_INST_IDX 2
+#elif defined(CONFIG_BT_RADIO_NOTIFICATION_CONN_CB_EGU_INST_EGU3)
+#define EGU_INST_IDX 3
+#elif defined(CONFIG_BT_RADIO_NOTIFICATION_CONN_CB_EGU_INST_EGU4)
+#define EGU_INST_IDX 4
+#elif defined(CONFIG_BT_RADIO_NOTIFICATION_CONN_CB_EGU_INST_EGU5)
+#define EGU_INST_IDX 5
+#elif defined(CONFIG_BT_RADIO_NOTIFICATION_CONN_CB_EGU_INST_EGU020)
+#define EGU_INST_IDX 020
+#else
+#error "Unknown EGU instance"
+#endif
+
+static nrfx_egu_t egu_inst = NRFX_EGU_INSTANCE(EGU_INST_IDX);
+static uint32_t configured_prepare_distance_us;
+static struct bt_radio_notification_conn_cb *registered_cb;
+
+static struct bt_conn *conn_pointers[CONFIG_BT_MAX_CONN];
+static uint8_t ppi_channels[CONFIG_BT_MAX_CONN];
+static struct k_work_delayable work_queue_items[CONFIG_BT_MAX_CONN];
+
+static uint32_t conn_interval_us_get(const struct bt_conn *conn)
+{
+	struct bt_conn_info info;
+
+	if (bt_conn_get_info(conn, &info)) {
+		return 0;
+	}
+
+	return BT_CONN_INTERVAL_TO_US(info.le.interval);
+}
+
+static void work_handler(struct k_work *work)
+{
+	uint8_t index = ARRAY_INDEX(work_queue_items, work);
+	struct bt_conn *conn = conn_pointers[index];
+
+	registered_cb->prepare(conn);
+
+	uint32_t conn_interval_us = conn_interval_us_get(conn);
+
+	/* Schedule the callback in the connection event does not occur because
+	 * of peripheral latency or scheduling conflicts.
+	 */
+	k_work_reschedule(&work_queue_items[index], K_USEC(conn_interval_us));
```

Comment:
What does happen after disconnection? It looks like this work is never canceled

Reply:
Fixed